servers/server.go

```diff
@@ -601,6 +611,138 @@ func (p *Server) Extract(source, destination string) error {
 	return files.Extract(p.GetFileServer(), source, destination, "*", false, nil)
 }
 
+func (p *Server) CreateBackup() (string, int64, error) {
+	sourceFiles := []string{p.GetFileServer().Prefix()}
+
+	backupDirectory := p.RunningEnvironment.GetBackupDirectory()
+	if backupDirectory == "" {
+		return "", 0, pufferpanel.ErrSettingNotConfigured("backupDirectory")
+	}
+
+	if p.GetEnvironment().IsBackingUp() {
```

Comment:
Technically a lock or something should be used, but this is still okay.

Reply:
I followed the pattern from Install, can add a lock here or leave it, happy with either